core-build/gulp-core-build-serve/src/ServeTask.ts

```diff
@@ -287,4 +288,66 @@ export class ServeTask<TExtendedConfig = {}> extends GulpTask<IServeTaskConfig &
       return undefined;
     }
   }
+
+  /**
+   * Workaround for loading gulp-connect, which automatically uses http2 if it
+   * can require() a module called 'http2'
+   *
+   * https://github.com/AveVlad/gulp-connect/issues/246
+   *
+   * In versions of NodeJS < 8:
+   *   'http2' would have to be a module in the node_modules folder.
+   *   We did not provide this normally, so most of the time nobody used http2 with gulp serve.
+   *
+   * However, in versions of NodeJS >= 8:
+   *   They provide a built-in module called 'http2', which is experimental and unstable
+   *   The built-in module is preferred unless an environment variable is set: NODE_NO_HTTP2=1
+   *
+   * We don't want to enforce environmental requirements, nor
+   * do we want to support a toolchain that relies on an experimental API.
+   *
+   * Until gulp-connect provides a way to disable HTTP2, we're using this workaround:
+   * Inject a falsey value into the require() cache, require gulp-connect,
+   * then restore the old cache state.
+   *
+   * As a consequence, this approach will prevent "gulp-connect" from using http2 in
+   * environments with Node < 8. This is intentional, because we don't see a reason to
+   * support http2 for serving localhost scripts.
+   */
+  // tslint:disable-next-line:no-any
+  private _loadGulpConnect(): any {
+    // this will raise an exception if it can't find http2,
+    // which happens if we are on Node6 and 'http2' has not been required yet
+    let http2CacheKey: string = 'http2';
+    try {
+      http2CacheKey = require.resolve('http2');
+    } catch (exception) {
+      // no-op
+    }
+
+    /* tslint:disable:typedef */
+    let gulpConnect;
+
+    // node 6 and http2 is in cache
+    if (Object.keys(require.cache).indexOf(http2CacheKey) !== -1) {
+      // store the old cache value
+      const http2CacheObject = require.cache[http2CacheKey];
+      require.cache[http2CacheKey] = { exports: undefined };
+
+      gulpConnect = require('gulp-connect');
+
+      // restore the old cache value
+      require.cache[http2CacheKey] = http2CacheObject;
```

Comment:
>require.cache[http2CacheKey] = http2CacheObject; [](start = 6, length = 48)

Ideally this would go in a try/finally block in case require() throws an exception, but maybe that's not realistic for our special case?